mne_icalabel/megnet/tests/test_label_components.py

```diff
@@ -0,0 +1,67 @@
+from unittest.mock import MagicMock
+
+import mne
+import numpy as np
+import onnxruntime as ort
+import pytest
+
+from mne_icalabel.megnet.label_components import (
+    _chunk_predicting,
+    _get_chunk_start,
+    megnet_label_components,
+)
+
+
+@pytest.fixture
+def raw_ica():
+    """Create a Raw instance and ICA instance for testing."""
+    sample_dir = mne.datasets.sample.data_path()
+    sample_fname = sample_dir / "MEG" / "sample" / "sample_audvis_raw.fif"
+
+    raw = mne.io.read_raw_fif(sample_fname).pick("mag")
+    raw.load_data()
+    raw.resample(250)
+    raw.notch_filter(60)
+    raw.filter(1, 100)
+
+    ica = mne.preprocessing.ICA(n_components=20, method="infomax", random_state=88)
+    ica.fit(raw)
+
+    return raw, ica
+
+
+def test_megnet_label_components(raw_ica):
+    """Test whether the function returns the correct artifact index."""
+    real_atrifact_idx = [0, 3, 5]  # heart beat, eye movement, heart beat
+    prob = megnet_label_components(*raw_ica)
+    this_atrifact_idx = list(np.nonzero(prob.argmax(axis=1))[0])
```

Comment:
It's a bit strange here. When I ran it yesterday, the model detection results were 0, 3, 5; but today they have become 0, 3, 4, 5, which caused the pytest to fail. It might be due to the randomness in the ICA algorithm？ (even though I set the random_state).

Reply:
Maybe we can try asserting `set(real_atrifact_idx).issubset(set(this_atrifact_idx))` to ensure that the true labels are a subset of the model's predicted labels. This is a debugging-specific approach, but it might not be entirely reasonable...

Reply:
If your ICA has a fix random state, and if the model is deterministic (which I think it is?), you should get the exact same results.

To debug you can try:
- run 2/3 times the 'icai' part and compare the mixing/unmixing matrixes with `assert_allclose`. They should be identical.
- run 2/3 times the model part and compare the model outputs (not the index, but the probabilities per class), they should be identical.